### DISCUSSION TITLE
Can't stop timer from REPL?  Keyboard interrupt does nothing?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [micropython/micropython#11256] [General]
Can't stop timer from REPL?  Keyboard interrupt does nothing?  I can't re-flash my ESP32, every tool I use says the ESP32 is busy.  The timers continue to run as and I can't seem to stop main.py at all.  Even keyboard interrupt fails to stop anything.

**Comment by Jibun-no-Kage** [comment]
Well, I finally got control back via a soft reset then CNTRL-B, oir was it CNTRL-A then CNTRL-B?  Either way the timers finally stopped, then I had to load os module and explicitly delete the main.py.

AM CALLING timer.deinit() for both timers on my ESP32 under the keyboard interrupt handler AND in the finally section, but deinit apparently DOES NOT stop timers.

I found a reference this bug?
https://github.com/micropython/micropython/issues/1113

**Comment by Jibun-no-Kage** [comment]
HOW do you stop timers?

**Comment by bixb922** [comment]
I don't know other ways to stop timers, but it is a good idea to start main.py with
```
import time
time.sleep(5)
```
That way you will have time to hit control-c just after reboot to recover access to REPL. Then you can enter
```
>>>import os
>>>os.rename("main.py","x.py")
```
and you have control back. This has helped me a couple of times.

**Comment by Jibun-no-Kage** [comment]
Yup, that is a good idea.  Using the CNTRL-A, CNTRL-B trick seems to work if I reboot and catch it.  The sleep gives you a much longer window as well.
